tests/js/blocks/content-helper/structure.test.js

```diff
@@ -4,20 +4,227 @@
 import {
 	render,
 	screen,
+	waitFor,
 } from '@testing-library/react';
 import '@testing-library/jest-dom';
 
 /**
  * Internal dependencies.
  */
 import RelatedTopPostList from '../../../../src/blocks/content-helper/components/related-top-post-list';
+import ContentHelperProvider, { RELATED_POSTS_DEFAULT_LIMIT, RELATED_POSTS_DEFAULT_TIME_RANGE } from '../../../../src/blocks/content-helper/content-helper-provider';
 
 describe( 'Content Helper', () => {
-	it( 'should display spinner when starting', () => {
+	test( 'should display spinner when starting', () => {
+		const getRelatedTopPostsMock = jest
+			.spyOn( ContentHelperProvider, 'getRelatedTopPosts' )
+			.mockImplementation( () => Promise.resolve( {} ) );
+
 		render( <RelatedTopPostList /> );
 
-		const spinner = screen.getByRole( 'presentation' );
+		expect( getRelatedTopPostsMock ).toHaveBeenCalled();
+
+		const spinner = getSpinner();
+		expect( spinner ).toBeInTheDocument();
 		expect( spinner ).toBeVisible();
-		expect( spinner ).toHaveClass( 'components-spinner' );
 	} );
+
+	test( 'should show contact us message when parsely site id is not set', async () => {
+		const getRelatedTopPostsMock = jest
+			.spyOn( ContentHelperProvider, 'getRelatedTopPosts' )
+			.mockImplementation( () => Promise.reject( {
+				errors: {
+					parsely_site_id_not_set: 'error msg',
+				},
+			} ) );
+
+		render( <RelatedTopPostList /> );
+		expect( getSpinner() ).toBeInTheDocument();
+
+		await waitFor( () => screen.findByTestId( 'parsely-contact-us' ), { timeout: 3000 } );
```

Comment:
Can we do this without passing timeout and just look for when the element appears on screen? ..... I tired multiple things but none of them worked

Reply:
Does it display on initial render or does it wait for a state update to render? If the latter you can look into using `act`

Reply:
By using act you mean this?

```
await act( async () => {
	waitFor( () => screen.findByTestId( 'parsely-contact-us' ) );
} );
```

It is not working while the above statement is working fine.